packages/carbon-web-components/src/components/side-panel/side-panel.ts

```diff
@@ -0,0 +1,1085 @@
+  currentStep;
+
+  /**
+   * Determines whether the side panel should render with an overlay
+   */
+  @property({ attribute: 'include-overlay', type: Boolean, reflect: true })
+  includeOverlay = false;
+
+  /**
+   * Sets the label text which will display above the title text
+   */
+  @property({ reflect: true, attribute: 'label-text' })
+  labelText;
+
+  /**
+   * Sets the icon description for the navigation back icon button
+   */
+  @property({ reflect: true, attribute: 'navigation-back-icon-description' })
+  navigationBackIconDescription = 'Back';
+
+  /**
+   * `true` if the side-panel should be open.
+   */
+  @property({ type: Boolean, reflect: true })
+  open = false;
+
+  /**
+   * SidePanel placement.
+   */
+  @property({ reflect: true })
+  placement = SIDE_PANEL_PLACEMENT.RIGHT;
+
+  /**
+   * Prevent closing on click outside of side-panel
+   */
+  @property({ type: Boolean, attribute: 'prevent-close-on-click-outside' })
+  preventCloseOnClickOutside = false;
+
+  /**
+   * Selector for page content, used to push content to side except
+   */
+  @property({ reflect: true, attribute: 'selector-page-content' })
+  selectorPageContent = '';
+
+  /**
+   * SidePanel size.
+   */
+  @property({ reflect: true })
+  size = SIDE_PANEL_SIZE.MEDIUM;
+
+  /**
+   * Determines if this panel slides in
+   */
+  @property({ attribute: 'slide-in', type: Boolean, reflect: true })
+  slideIn = false;
+
+  // /**
+  //  * Provide a `Slug` component to be rendered inside the `SidePanel` component
+  //  */
+  // @property({ reflect: true })
+  // slug;
+
+  /**
+   * Sets the subtitle text
+   */
+  @property({ reflect: false })
+  title;
+
+  async connectObservers() {
+    await this.updateComplete;
+    this._hObserveResize = observeResize(this._resizeObserver, this._sidePanel);
+    this._hObserveActionsResize = observeResize(
+      this._resizeActionsObserver,
+      this._actions
+    );
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.disconnectedCallback();
+    this.connectObservers();
+  }
+
+  disconnectedCallback() {
+    if (this._hObserveResize) {
+      this._hObserveResize = this._hObserveResize.release();
+    }
+    if (this._hObserveActionsResize) {
+      this._hObserveActionsResize = this._hObserveActionsResize.release();
+    }
+
+    if (this._innerContent) {
+      this._innerContent.removeEventListener('scroll', this._scrollObserver);
+    }
+  }
+
+  render() {
+    const {
+      animateTitle,
+      closeIconDescription,
+      condensedActions,
+      currentStep,
+      includeOverlay,
+      labelText,
+      navigationBackIconDescription,
+      open,
+      placement,
+      size,
+      slideIn,
+      title,
+    } = this;
+
+    if (!open && !this._isOpen) {
+      return html``;
+    }
+
+    const containerClass = this.containerClass
+      .split(' ')
+      .filter(Boolean)
+      .reduce((acc, item) => ({ ...acc, [item]: true }), {});
+
+    const containerClasses = classMap({
```

Comment:
one thing different about web components is since now we have `cds-side-panel` we can leverage that and avoid using all of these different classes at the root of the component. Instead we would have props like `size`, and in styles do something like 

```
:host(cds-side-panel[size='lg']){
  @extend .${blockClass}__container--lg;
}
```

accordion to be a good component to look at which also has something similar to opening/closing. In accordion `expanding` / `collapsing` are being used

Reply:
@ariellalgilmore would you apply that to philosophy to the internals or just the component?

For example, is it preferable to add attributes rather the ibm-products classes, simpler classes to the title-container?

So 
```
.side-panel__on-detail-step <-- added to HTML or 

// Shorter CSS class inside component
.detail-step {
   @extend .side-panel__on-detail-step
}

// ID + attribute
#title-container[detail-step] {
  @extend .side-panel__on-detail-step
}
```

Reply:
for the internal classes you can keep the same classes as ibm-products, but there shouldn't be any container classes for components. Instead the container classes are usually used as props